File: frontend/lib/features/habits/domain/entities/habit.dart
import 'dart:collection';

class Habit {
  String id;
  Map<String, String> name;
  String categoryId;
  bool reviewed;
  Map<String, String> description;
  String icon;
  HashSet<String> unitIds;

  Habit({
    required this.id,
    required this.name,
    required this.categoryId,
    required this.reviewed,
    required this.description,
    required this.icon,
    required this.unitIds,
  });
}

File: backend/src/features/habits/structs/requests/habit.rs
use std::collections::{HashMap, HashSet};

use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Deserialize)]
pub struct GetHabitParams {
    pub habit_id: Uuid,
}

#[derive(Deserialize)]
pub struct UpdateHabitParams {
    pub habit_id: Uuid,
}

#[derive(Deserialize)]
pub struct MergeHabitsParams {
    pub habit_to_delete_id: Uuid,
    pub habit_to_merge_on_id: Uuid,
}

#[derive(Deserialize, Serialize)]
pub struct HabitUpdateRequest {
    pub name: HashMap<String, String>,
    pub description: HashMap<String, String>,
    pub category_id: Uuid,
    pub reviewed: bool,
    pub icon: String,
    pub unit_ids: HashSet<Uuid>,
}

#[derive(Deserialize, Serialize)]
pub struct HabitCreateRequest {
    pub name: HashMap<String, String>,
    pub description: HashMap<String, String>,
    pub category_id: Uuid,
    pub icon: String,
    pub unit_ids: HashSet<Uuid>,
}
